op.c

```diff
@@ -16651,24 +16720,21 @@ Perl_subsignature_append_slurpy(pTHX_ I32 sigil, PADOFFSET padix)
     if(signature->slurpy)
         yyerror("Multiple slurpy parameters not allowed");
 
-    UV argix = signature->elems;
+    UV argix = signature->next_argix;
+    /* do not increment */
 
     signature->slurpy = (char)sigil;
 
-    if(padix) {
-        assert(PadnamePV(PadnamelistARRAY(PL_comppad_name)[padix])[0] == sigil);
+    if(!padix)
+        return;
 
-        OP *varop = newUNOP_AUX(OP_ARGELEM, 0, NULL, NULL);
-        if(sigil == '@')
-            varop->op_private |= OPpARGELEM_AV;
-        if(sigil == '%')
-            varop->op_private |= OPpARGELEM_HV;
-        varop->op_targ = padix;
-        cUNOP_AUXx(varop)->op_aux = INT2PTR(UNOP_AUX_item *, argix);
+    struct yy_parser_signature_param *param = subsignature_push_param();
 
-        signature->elemops = op_append_list(OP_LINESEQ, signature->elemops,
-                newSTATEOP(0, NULL, varop));
-    }
+    param->argix = argix;
+
+    assert(PadnamePV(PadnamelistARRAY(PL_comppad_name)[padix])[0] == sigil);
+
+    param->padix = padix;
```

Comment:
Add a `/* this is a different struct member */` or `/*different var */` comment, or maybe rename `padix` the c auto to `pad_ix`. these 2 lines took too many seconds with eyeballs to figure out they arent the same line accidentally written twice. makes it easier to read by future ppl.